doc/akri_connector/connector-metadata-schema.json

```diff
@@ -0,0 +1,305 @@
+        },
+        "secretKey": {
+          "type": "string",
+          "description": "The key in the secret to be mounted"
+        },
+        "secretRef": {
+          "type": "string",
+          "description": "The name of the secret to be mounted"
+        }
+      },
+      "required": [
+        "secretAlias",
+        "secretKey",
+        "secretRef"
+      ]
+    },
+    "architecture": {
+      "type": "string",
+      "description": "The architecture that this connector should run on"
+    },
+    "sourceCode": {
+      "type": "object",
+      "description": "Details about the source code that compiled this connector",
+      "properties": {
+        "language": {
+          "type": "string",
+          "description": "The programming language used to write this connector"
+        },
+        "languageVersion": {
+          "type": "string",
+          "description": "The version of the programming language used to write this connector"
+        },
+        "sdk": {
+          "type": "object",
+          "description": "Details about the versions of the AIO SDKs used to write this connector (if they were used)",
+          "properties": {
+            "mqttPackageVersion": {
+              "type": "string",
+              "description": "The version of the azure iot operations MQTT package"
+            },
+            "servicesPackageVersion": {
+              "type": "string",
+              "description": "The version of the azure iot operations services package"
+            },
+            "connectorPackageVersion": {
+              "type": "string",
+              "description": "The version of the azure iot operations connector package"
+            }
+          }
+        }
+      },
+      "required": [
+        "language",
+        "languageVersion"
+      ]
+    },
+    "resourceRequirements": {
+      "type": "object",
+      "description": "The minimal resource requirements to run this connector",
+      "properties": {
+        "cpu": {
+          "type": "string"
+        },
+        "memory": {
+          "type": "string"
+        },
+        "storage": {
+          "type": "string"
+        }
+      }
+    },
+    "aioMetadata": {
+      "type": "object",
+      "description": "The AIO requirements to run this connector",
+      "properties": {
+        "aioMinVersion": {
+          "type": "string",
+          "description": "The minimal (inclusive) version that can run this connector."
+        },
+        "aioMaxVersion": {
+          "type": "string",
+          "description": "The maximal (inclusive) version that can run this connector."
+        }
+      }
+    },
+    "inboundEndpoints": {
+      "type": "array",
+      "description": "The array of inbound endpoints that this connector connects with.",
+      "items": [
+        {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string"
+            },
```

Comment:
I thought you want to version the endpointTypes? It is called endpointType in Device.endpoints.inbound and we should be use similar naming. 
Should we also add a comment that shows the recommended (required?) format like <companyname>.<protocol>

Reply:
I'll rename this to "endpointType", sure. I don't believe we have a recommended or required format, but maybe @abhipsaMisra can comment on that.